internal/controller/openslo/slo_controller.go

```diff
@@ -119,16 +119,17 @@ func (r *SLOReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		log.Info("PrometheusRule not found. Let's make one.")
 		prometheusRule, err = helpers.CreatePrometheusRule(slo, sli)
 		if err != nil {
-			err = utils.UpdateStatus(ctx, slo, r.Client, "Ready", metav1.ConditionFalse, "Failed to create Prometheus Rule")
-			if err != nil {
-				log.Error(err, "Failed to update SLO status")
-				return ctrl.Result{}, err
+			r.Recorder.Event(slo, "Warning", "FailedToCreatePrometheusRule", "Failed to create Prometheus Rule")
+			errUpdateStatus := utils.UpdateStatus(ctx, slo, r.Client, "Ready", metav1.ConditionFalse, fmt.Sprintf("Failed to create new Prometheus Rule: %v", err))
+			if errUpdateStatus != nil {
```

Comment:
I'm not sure I'm quite getting the reasons behind this `errUpdateState` thing.

Reply:
What was happening before is that you had an err value that was being overwritten by another assignment generated by `utils.UpdateStatus`, therefore [this code](https://github.com/oskoperator/osko/pull/77/files/060f4b69b3d02e8b98393e3f9cdad6e6f73ebbc9#diff-843e6e74bf74fb2516f330ec7542781819a468e33325a9e82465e0524ad980d5L127-L128) never executed even though it should've